src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  static updateAliveAnimals() {
+    this.alive = this.alive.filter(animal => animal.health > 0);
+  };
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
-}
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
```

Comment:
The hide method always sets the hidden property to true, which means that once a Herbivore is hidden, it can never unhide. Consider changing the implementation to toggle the hidden state, or perhaps add a separate method for unhiding.